rust/noosphere-core/src/tracing.rs

```diff
@@ -1,27 +1,390 @@
 ///! Helpers to wrangle logging across Noosphere crates
 ///! NOTE: [initialize_tracing] should only ever be called in tests or binaries;
 ///! a library should only concern itself with instrumentation and logging.
-use std::sync::Once;
+use strum_macros::{Display, EnumString};
 
-static INITIALIZE_TRACING: Once = Once::new();
+/// The crates that are considered when evaluating [NoosphereLog] and
+/// [NoosphereLogLevel] as directive configuration.
+pub static NOOSPHERE_LOG_LEVEL_CRATES: &[&str] = &[
+    "noosphere",
+    "noosphere_core",
+    "noosphere_storage",
+    "noosphere_sphere",
+    "noosphere_into",
+    "noosphere_gateway",
+    "noosphere_collections",
+    "noosphere_cli",
+    "noosphere_car",
+    "noosphere_api",
+    "noosphere_ns",
+    "tower_http",
```

Comment:
Any ideas on keeping this in sync with new crates? Could be easy to miss

Reply:
No, open to suggestions